Contents/mods/More Traits/media/lua/client/MoreTraits.lua

```diff
@@ -1270,28 +1282,31 @@ local function amputee(_player)
         local ForeArm_L = bodydamage:getBodyPart(BodyPartType.FromString("ForeArm_L"));
         local Hand_L = bodydamage:getBodyPart(BodyPartType.FromString("Hand_L"));
         if UpperArm_L:HasInjury() then
-            UpperArm_L:SetBitten(false);
-            UpperArm_L:setScratched(false);
-            UpperArm_L:setDeepWounded(false);
-            UpperArm_L:setBleeding(false);
-            UpperArm_L:setHaveGlass(false);
-            UpperArm_L:SetInfected(false);
+            UpperArm_L:RestoreToFullHealth();
+            if justGotInfected then
+                player:getBodyDamage():setInfected(false);
+            end
+            player:getVisual():setBlood(BloodBodyPartType.FromIndex(BodyPartType.ToIndex(UpperArm_L:getType())), 0);
+            player:getVisual():setDirt(BloodBodyPartType.FromIndex(BodyPartType.ToIndex(UpperArm_L:getType())), 0);
+            player:resetModelNextFrame();
         end
         if ForeArm_L:HasInjury() then
-            ForeArm_L:SetBitten(false);
-            ForeArm_L:setScratched(false);
-            ForeArm_L:setDeepWounded(false);
-            ForeArm_L:setBleeding(false);
-            ForeArm_L:setHaveGlass(false);
-            ForeArm_L:SetInfected(false);
+            ForeArm_L:RestoreToFullHealth();
+            if justGotInfected then
+                player:getBodyDamage():setInfected(false);
+            end
+            player:getVisual():setBlood(BloodBodyPartType.FromIndex(BodyPartType.ToIndex(ForeArm_L:getType())), 0);
+            player:getVisual():setDirt(BloodBodyPartType.FromIndex(BodyPartType.ToIndex(ForeArm_L:getType())), 0);
+            player:resetModelNextFrame();
```

Comment:
Will resetting the model cause the player to reset all animation states? It might cause unforseen issues.

Reply:
Just checked with exercises and building, at least with these activities this code does interrupt animations. 
If you fear unforseen, this is not the most important feature to be keeped.